src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+                ]
+            else:
+                filepath_list = [
+                    self._get_raw_path(year=year, month=month, day=day)
+                    for (year, month, day) in date_list
+                ]
+        filepath_list.sort()
+
+        # Merge all files in one dataset
+        if len(filepath_list) == 0:
+            raise FileNotFoundError(
+                f"Cannot find any netcdf file in the form"
+                f"{processed_path}. Make "
+                f"sure that you have already called the 'process' method."
+            )
+        else:
+            try:
+                ds = xr.open_mfdataset(
+                    filepath_list,
+                    decode_times=False,
+                )
+                # include the names of all files that are included in the ds
+                ds.attrs["included_files"] = [f.stem for f in filepath_list]
+            except FileNotFoundError:
+                raise FileNotFoundError(
+                    "Cannot find one or more netcdf files corresponding "
+                    "to the selected range. Make sure that you already "
+                    "downloaded and processed those data."
+                )
+
+        return ds.rio.write_crs("EPSG:4326", inplace=True)
+
+    def _download(self, year, month, day, clobber):
+        # Preparatory steps for the actual download
+        output_filepath = self._get_raw_path(year=year, month=month, day=day)
+        output_filepath.parent.mkdir(parents=True, exist_ok=True)
+        url = self._get_url(year=year, month=month, day=day)
+        # Actual download
+        return _actual_download(
+            filepath=output_filepath,
+            url=url,
+            clobber=clobber,
+        )
+
+    def _create_date_list(
+        self, start_year, end_year, start_month, end_month, start_day, end_day
+    ):
+
+        # Check whether the input dates are valid, assign dates when not
+        # specified and create list of tuples containing the range of
+        # dates of interest
+
+        if (
+            start_month is not None or start_day is not None
+        ) and start_year is None:
+            raise ValueError(
+                "If you specify starting month or day, you also "
+                "need to specify the starting year."
+            )
+
+        if (end_month is not None or end_day is not None) and end_year is None:
+            raise ValueError(
+                "If you specify end month or day, you also "
+                "need to specify the end year."
+            )
+
+        if start_day is not None and start_month is None:
+            raise ValueError(
+                "If you specify the starting day, you also "
+                "need to specify the starting month."
+            )
+
+        if end_day is not None and end_month is None:
+            raise ValueError(
+                "If you specify the end day, you also "
+                "need to specify the end month."
+            )
```

Comment:
not sure if this would overcomplicate things, but there might be an easy way to make two tuples of ("starting", start_month, start_day, and start_year) and the same for the "end" pars. And then you don't have to duplicate the text?

Reply:
I'd prefer to keep it this way, as I think it's more clear for the user